cpp/src/examples/image_classifier/resnet-18/resnet-18_handler.cc

```diff
@@ -0,0 +1,261 @@
+
+        const cv::Rect roi(offsetW, offsetH, cropSize, cropSize);
+        image = image(roi);
+
+        // Convert the image to GPU Mat
+        cv::cuda::GpuMat gpuImage;
+        cv::Mat resultImage;
+
+        gpuImage.upload(image);
+
+        // Resize on GPU
+        cv::cuda::resize(gpuImage, gpuImage,
+                         cv::Size(kTargetImageSize, kTargetImageSize));
+
+        // Convert to BGR on GPU
+        cv::cuda::cvtColor(gpuImage, gpuImage, cv::COLOR_BGR2RGB);
+
+        // Convert to float on GPU
+        gpuImage.convertTo(gpuImage, CV_32FC3, 1 / 255.0);
+
+        // Download the final image from GPU to CPU
+        gpuImage.download(resultImage);
+
+        // Create a tensor from the CPU Mat
+        torch::Tensor tensorImage = torch::from_blob(
+            resultImage.data, {resultImage.rows, resultImage.cols, 3},
+            torch::kFloat);
+        tensorImage = tensorImage.permute({2, 0, 1});
+
+        std::vector<double> norm_mean = {kImageNormalizationMeanR,
+                                         kImageNormalizationMeanG,
+                                         kImageNormalizationMeanB};
+        std::vector<double> norm_std = {kImageNormalizationStdR,
+                                        kImageNormalizationStdG,
+                                        kImageNormalizationStdB};
+
+        // Normalize the tensor
+        tensorImage = torch::data::transforms::Normalize<>(
+            norm_mean, norm_std)(tensorImage);
+
+        tensorImage.clone();
+        batch_tensors.emplace_back(tensorImage.to(*device));
+        idx_to_req_id.second[idx++] = request.request_id;
+      } else if (dtype_it->second == "List") {
+        // case3: the image is a list
+      }
+    } catch (const std::runtime_error& e) {
+      TS_LOGF(ERROR, "Failed to load tensor for request id: {}, error: {}",
+              request.request_id, e.what());
+      auto response = (*response_batch)[request.request_id];
+      response->SetResponse(500, "data_type",
+                            torchserve::PayloadType::kDATA_TYPE_STRING,
+                            "runtime_error, failed to load tensor");
+    } catch (const c10::Error& e) {
+      TS_LOGF(ERROR, "Failed to load tensor for request id: {}, c10 error: {}",
+              request.request_id, e.msg());
+      auto response = (*response_batch)[request.request_id];
+      response->SetResponse(500, "data_type",
+                            torchserve::PayloadType::kDATA_TYPE_STRING,
+                            "c10 error, failed to load tensor");
+    }
+  }
+  if (!batch_tensors.empty()) {
+    batch_ivalue.emplace_back(torch::stack(batch_tensors).to(*device));
+  }
+
+  return batch_ivalue;
+}
+
+void ResnetHandler::Postprocess(
+    const torch::Tensor& data,
+    std::pair<std::string&, std::map<uint8_t, std::string>&>& idx_to_req_id,
+    std::shared_ptr<torchserve::InferenceResponseBatch>& response_batch) {
+  std::ifstream jsonFile("index_to_name.json");
```

Comment:
Can you move loading of the file into the constructor to avoid reloading every time we do postprocessing?